agent/eventhandler/task_handler_types.go

```diff
@@ -254,3 +257,17 @@ func updateContainerSentStatus(container *apicontainer.Container, status apicont
 		}
 	}
 }
+
+func updateManagedAgentSentStatus(container *apicontainer.Container, managedAgentName string, status apicontainerstatus.ManagedAgentStatus, dataClient data.Client) {
+	saveContainer := false
```

Comment:
do we need this variable? why not just save container inside the below `if`?

Reply:
This follows the pattern in updateContainerSentStatus above.  I like the separation of these two processes -- update/save.

Reply:
Hmm, I also think this variable is unnecessary. This is more concise, and reads better IMO (it's still 2 processes)

```golang
	if container.GetManagedAgentSentStatus(managedAgentName) != status {
		container.UpdateManagedAgentSentStatus(managedAgentName, status)
		if err := dataClient.SaveContainer(container); err != nil {
			seelog.Errorf("Failed to update %s managed agent sent status in database for container %s: %v", managedAgentName, container.Name, err)
		}
	}
```